FWCore/Framework/src/WorkerRegistry.cc

```diff
@@ -44,4 +44,23 @@ namespace edm {
     }
     return (workerIt->second.get());
   }
+
+  Worker const* WorkerRegistry::get(std::string const& moduleLabel) const {
+    WorkerMap::const_iterator workerIt = m_workerMap.find(moduleLabel);
+    if (workerIt != m_workerMap.end()) {
+      return workerIt->second;
+    }
+    return nullptr;
+  }
+
+  void WorkerRegistry::deleteModule(std::string const& moduleLabel) {
+    WorkerMap::iterator workerIt = m_workerMap.find(moduleLabel);
+    if (workerIt == m_workerMap.end()) {
+      throw cms::Exception("LogicError")
+          << "WorkerRegistry::deleteModule() Trying to delete the module of a Worker with label " << moduleLabel
+          << " but no such Worker exists in the WorkerRegistry. Please contact framework developers.";
+    }
+    workerIt->second->clearModule();
```

Comment:
@Dr15Jones Actually in a quick test with
```suggestion
    m_workerMap.erase(workerIt);
```
the (relevant) test configurations run. That would leave stale pointers to the `Worker` though e.g. in all ProductResolvers corresponding the products of the deleted producers. Would it be good to try to clean up all those as well?

Reply:
I really hate leaving stale pointers anywhere.

Reply:
Thanks (my thoughts exactly), so I will not commit the change above and look into cleaning up the objects that would end up having stale pointers instead.